ddtrace/tracer.py

```diff
@@ -670,18 +670,11 @@ def _initialize_span_processors(self, appsec_enabled=asbool(get_env("appsec", "e
         trace_processors += [TraceTopLevelSpanProcessor()]
         trace_processors += self._filters
 
-        self._span_processors = [
-            SpanAggregator(
-                partial_flush_enabled=self._partial_flush_enabled,
-                partial_flush_min_spans=self._partial_flush_min_spans,
-                trace_processors=trace_processors,
-                writer=self._writer,
-            ),
-        ]  # type: List[SpanProcessor]
+        self._span_processors = []  # type: List[SpanProcessor]
 
         if appsec_enabled:
             try:
-                from .appsec.processor import AppSecSpanProcessor
+                from .appsec._processor import AppSecSpanProcessor
```

Comment:
I might have overlooked this in past PRs, but I was wondering if the tracer could implement some API to allow other components to interact with it. This current implementation seems to crate a tight bond between the tracer and AppSec (i.e. the tracer effectively knows about the `.appsec` module, even if there is a runtime check). This could be broken if the tracer allowed for some generic hooking mechanism, so that it can go on without knowing that `.appsec` exists. IMHO it's AppSec that should know about the tracer, and not the other way around. WDYT?

Reply:
I'd be +1 on that. At some point, doubling down on the gateway should bring us here.
However, we'll need to find a way to abstract writing on spans from these plugins.
Let's fix the current bug and move forward :)

Reply:
Yeah the idea was/is to expose some kind of API for adding/removing/specifying span processors we just haven't decided on a public API for this so we're keeping things internal to the tracer.

I think right now things are mostly contained in `_initialize_span_processors` so would be quite easily translated to whatever API we decide on.